modules/statistics/AvgRTPStatsReporter.js

```diff
@@ -75,6 +75,205 @@ class AverageStatReport {
 }
 
 /**
+ * Class gathers the stats that are calculated and reported for a
+ * {@link TraceablePeerConnection} even if it's not currently active. For
+ * example we want to monitor RTT for the JVB connection while in P2P mode.
+ */
+class ConnectionAvgStats {
+    /**
+     * Creates new <tt>ConnectionAvgStats</tt>
+     * @param {JitsiConference} conference
+     * @param {boolean} isP2P
+     * @param {number} n the number of samples, before arithmetic mean is to be
+     * calculated and values submitted to the analytics module.
+     */
+    constructor(conference, isP2P, n) {
+        /**
+         * Is this instance for JVB or P2P connection ?
+         * @type {boolean}
+         */
+        this.isP2P = isP2P;
+
+        /**
+         * How many samples are to be included in arithmetic mean calculation.
+         * @type {number}
+         * @private
+         */
+        this._n = n;
+
+        /**
+         * The current sample index. Starts from 0 and goes up to {@link _n})
+         * when analytics report will be submitted.
+         * @type {number}
+         * @private
+         */
+        this._sampleIdx = 0;
+
+        /**
+         * Average round trip time reported by the ICE candidate pair.
+         * @type {AverageStatReport}
+         */
+        this._avgRTT = new AverageStatReport('stat.avg.rtt');
+
+        /**
+         * Map stores average RTT to the JVB reported by remote participants.
+         * Mapped per participant id {@link JitsiParticipant.getId}.
+         *
+         * This is used only when {@link ConnectionAvgStats.isP2P} equals to
+         * <tt>false</tt>.
+         *
+         * @type {Map<string,AverageStatReport>}
+         * @private
+         */
+        this._avgRemoteRTTMap = new Map();
+
+        /**
+         * The conference for which stats will be collected and reported.
+         * @type {JitsiConference}
+         * @private
+         */
+        this._conference = conference;
+
+        this._onConnectionStats = (tpc, stats) => {
+            if (this.isP2P === tpc.isP2P) {
+                this._calculateAvgStats(stats);
+            }
+        };
+        conference.statistics.addConnectionStatsListener(
+            this._onConnectionStats);
+
+        if (!this.isP2P) {
+            this._onUserLeft = id => this._avgRemoteRTTMap.delete(id);
+            conference.on(ConferenceEvents.USER_LEFT, this._onUserLeft);
+
+            this._onRemoteStatsUpdated
+                = (id, data) => this._processRemoteStats(id, data);
+            conference.on(
+                ConnectionQualityEvents.REMOTE_STATS_UPDATED,
+                this._onRemoteStatsUpdated);
+        }
+    }
+
+    /**
+     * Processes next batch of stats.
+     * @param {go figure} data
```

Comment:
I thought the convention was that the names alternate -- "data" from one function goes to "object" in the next function, and then back to "data" ;)